src/mixins/ERC4626.sol

```diff
@@ -59,48 +59,44 @@ abstract contract ERC4626 is ERC20 {
         afterDeposit(underlyingAmount);
     }
 
-    function withdraw(address to, uint256 underlyingAmount) public virtual returns (uint256 shares) {
-        _burn(msg.sender, shares = calculateShares(underlyingAmount));
+    function mint(address to, uint256 shareAmount) public virtual returns (uint256 underlyingAmount) {
+        _mint(to, shareAmount);
```

Comment:
I think this is incorrect, this should be `_mint(to, underlyingAmount = calculateUnderlying(shareAmount)`.
In the current implementation it always returns an `underlyingAmount` of 0.